src/vm/hints/keccak_utils.rs

```diff
@@ -109,9 +109,130 @@ pub fn unsafe_keccak(
     }
 }
 
+/*
+Implements hint:
+
+    %{
+        from eth_hash.auto import keccak
+        keccak_input = bytearray()
+        n_elms = ids.keccak_state.end_ptr - ids.keccak_state.start_ptr
+        for word in memory.get_range(ids.keccak_state.start_ptr, n_elms):
+            keccak_input += word.to_bytes(16, 'big')
+        hashed = keccak(keccak_input)
+        ids.high = int.from_bytes(hashed[:16], 'big')
+        ids.low = int.from_bytes(hashed[16:32], 'big')
+    %}
+
+ */
+pub fn unsafe_keccak_finalize(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    /* -----------------------------
+    Just for reference (cairo code):
+    struct KeccakState:
+        member start_ptr : felt*
+        member end_ptr : felt*
+    end
+    ----------------------------- */
+
+    let keccak_state_ptr =
+        match get_address_from_var_name("keccak_state", &ids, vm, hint_ap_tracking) {
+            Ok(MaybeRelocatable::RelocatableValue(relocatable)) => relocatable,
+            Err(e) => return Err(e),
+            _ => unreachable!(),
```

Comment:
Is it actually `unreachable!`? What if I mistakenly stored a number in a variable I called `keccak_state`? Otherwise, if `get_address_from_var_name` is guaranteed to return an address, why doesn't it return `Relocatable` directly?